src/weierstrass/curve.rs

```diff
@@ -76,6 +80,73 @@ impl EllipticCurve for Curve {
             z: self.f.one(),
         })
     }
+    fn serialize(&self, p: &Self::Point, compress: bool, buf: &mut [u8]) {
+        let p_coords = &p.c;
+        let x = &p_coords.x;
+        let y = &p_coords.y;
+        let mut x_bytes = x.to_bytes_be();
+        let mut y_bytes = y.to_bytes_be();
+        let out_bytes = match compress {
+            true => {
+                let choice = (y.sgn0_le() > 0) as u8;
+                // if sign > 0: tag = 0x02; elif sign < 0: tag = 0x03;
+                let tag = u8::conditional_select(&0x02, &0x03, choice.into());
+                // sign should be 1/-1, so this cast should work
+                let mut o = vec![tag];
+                o.append(&mut x_bytes);
+                o
+            },
+            _ => {
+                let mut o: Vec<u8> = vec![0x04];
+                o.append(&mut x_bytes);
+                o.append(&mut y_bytes);
+                o
+            }
+        };
+        buf.copy_from_slice(&out_bytes);
+    }
+    fn deserialize(&self, buf: &[u8]) -> Result<Self::Point,Error> {
+        let p = self.f.get_modulus();
+        let max_bytes = (p.bits()+7)/8;
+        let tag = buf[0];
+        match tag {
+            0x04 => {
+                if buf.len() != 2*max_bytes+1 {
+                    return Err(Error::new(ErrorKind::Other, "Invalid bytes for deserialization"));
+                }
+                let x = self.f.elt(BigInt::from_bytes_be(Sign::Plus, &buf[1..max_bytes+1]));
+                let y = self.f.elt(BigInt::from_bytes_be(Sign::Plus, &buf[max_bytes+1..]));
+                Ok(self.new_point(ProyCoordinates {
+                    x: x,
+                    y: y,
+                    z: self.f.one()
+                }))
+            }
+            0x02 | 0x03 => {
+                if buf.len() != max_bytes+1 {
+                    return Err(Error::new(ErrorKind::Other, "Invalid bytes for deserialization"));
+                }
+                // recompute y coordinate
+                let x = self.f.elt(BigInt::from_bytes_be(Sign::Plus, &buf[1..max_bytes+1]));
+                let xx = &x * &x;
+                let xx_a = &xx + &self.a;
+                let xxx_ax = &xx_a * &x;
+                let xxx_ax_b = &xxx_ax + &self.b;
+                let y_sqrt = xxx_ax_b.sqrt();
+                let y_sgn_bit = (y_sqrt.sgn0_le() == -1) as u8;
```

Comment:
Use the function `tag : u8 = ((s>>1)&0x1)+2 ` for getting `tag` from `s`. then compare both tags. you can do  `y=-y`.